src/nunit.analyzers/DiagnosticSuppressors/DereferencePossiblyNullReferenceSuppressor.cs

```diff
@@ -61,38 +61,116 @@ private static bool ShouldBeSuppressed(SyntaxNode node)
                 // Drop the cast.
                 possibleNullReference = castExpression.Expression.ToString();
             }
+            else if (node.IsKind(SyntaxKind.CoalesceExpression) && node is BinaryExpressionSyntax binaryExpression)
+            {
+                // The compiler complains about the whole expression instead of just the right operand.
+                possibleNullReference = binaryExpression.Right.ToString();
+            }
+            else if (node is ConditionalExpressionSyntax conditionalExpression)
+            {
+                // The compiler complains about the whole expression instead of just one of the operands.
+                // Try to determine which one.
+                ExpressionSyntax? conditionalExpressionPath = DetermineConditionalExpressionPath(conditionalExpression);
+
+                if (conditionalExpressionPath is null)
+                {
+                    // We don't know.
+                    return false;
+                }
+
+                possibleNullReference = conditionalExpressionPath.ToString();
+            }
 
             for (SyntaxNode? currentNode = node; currentNode is not null;)
             {
-                StatementSyntax? statement = currentNode.Ancestors().OfType<StatementSyntax>().FirstOrDefault();
+                StatementSyntax? statement = currentNode.AncestorsAndSelf().OfType<StatementSyntax>().FirstOrDefault();
 
                 if (statement is null)
                 {
                     break;
                 }
 
-                BlockSyntax? parent = statement.Ancestors().OfType<BlockSyntax>().FirstOrDefault();
+                BlockSyntax? block = statement.Parent as BlockSyntax;
 
                 // If the statement is inside an Assert.Multiple NUnit doesn't throw
                 // flow continues with actual null values and therefore these shouldn't be suppressed.
                 if (!AssertHelper.IsInsideAssertMultiple(statement))
                 {
                     if (IsKnownToBeNotNull(currentNode) ||
-                        (parent is not null && IsValidatedNotNullByPreviousStatementInSameBlock(possibleNullReference, parent, statement)))
+                        (block is not null && IsValidatedNotNullByPreviousStatementInSameBlock(possibleNullReference, block, statement)))
                     {
                         return true;
                     }
                 }
 
-                currentNode = parent;
+                currentNode = statement.Parent;
```

Comment:
Is there a reason that we don't use `parent` instead of `statement.Parent`

Reply:
The `parent` variable no longer exists. It is now called `block` and when a statement is not inside a block (which is the use case that was previously not covered), `block` is now `null`. Previously it would keep looking to parents until it found a block.

Given:
```csharp
{
    Assert.That(possible-nullable-value, Is.Not.Null);
    if (condition)
       non-nullable-variable = possible-nullable-value;
}
```

The offending statement is the assignment operation.

The `parent` previously was the block containing the `if`. 
The method `IsValidatedNotNullByPreviousStatementInSameBlock` tries to find the assignment statement inside that block, but it isn't in there as a direct statement and therefore doesn't find it and hence doesn't look to previous statements for assertions.

Now the `block` is `null` and the code moves on to `statement.Parent` which is the `if` statement. It then find the block and in there that `if` statement and before that the `Assert`.

Reply:
My mistake I meant `block` and not `parent` in the question. Thanks for the thorough explanation, as far as I can tell it makes sense.